server/proxy/install.go

```diff
@@ -176,12 +173,34 @@ func (p *Proxy) ensureBot(manifest *apps.Manifest, actingUserID string, client *
 		}
 	}
 
-	token, response := client.CreateUserAccessToken(fullBot.UserId, "Mattermost App Token")
-	if response.StatusCode != http.StatusOK {
-		if response.Error != nil {
-			return nil, nil, response.Error
+	var token *model.UserAccessToken
+	if manifest.RequestedPermissions.Contains(apps.PermissionActAsBot) {
+		user, response = client.GetUser(fullBot.UserId, "")
+		if user == nil {
+			if response.Error != nil {
+				return nil, nil, errors.Wrap(response.Error, "failed to get bot user")
+			}
+			return nil, nil, errors.Errorf("failed to get bot user, status code = %v", response.StatusCode)
+		}
+
+		if !strings.Contains(user.Roles, model.SYSTEM_POST_ALL_ROLE_ID) {
+			newRoles := user.Roles + " " + model.SYSTEM_POST_ALL_ROLE_ID
+			updated, res := client.UpdateUserRoles(fullBot.UserId, newRoles)
+			if !updated {
+				if res.Error != nil {
+					return nil, nil, errors.Wrap(res.Error, "failed to update bot user's roles")
+				}
+				return nil, nil, errors.Errorf("failed to update bot user's roles, status code = %v", res.StatusCode)
+			}
+		}
```

Comment:
While I don't have objections to this change, it should be clearly documented (maybe even as part of the UX) that bots can read all posts in all channels. Surely, that can be a separate ticket.